integration/spark2/src/main/scala/org/apache/spark/util/MergeIndexUtil.scala

```diff
@@ -52,9 +52,17 @@ object MergeIndexUtil {
                      CarbonCommonConstants.LOAD_FOLDER.length)
         mergedSegmentIds.add(loadName)
       })
+      val loadFolderDetailsArray = SegmentStatusManager
```

Comment:
Avoid reading the table status file. CompactionModel already holds the loaddetails

Reply:
compactionModel holds the details of the already existing segemnts. not the newly compacted segments. so need to read it again